src/test/scala/uk/gov/nationalarchives/omega/jms/JmsRRSqsClientSpec.scala

```diff
@@ -0,0 +1,130 @@
+    // Create a Jms Request-Reply Client (e.g. from the Play Application start up method)
+    val clientRes: Resource[IO, JmsRRClient[IO]] = JmsRRClient.createForSqs[IO](
+      HostBrokerEndpoint("localhost", 9324),
+      UsernamePasswordCredentials("x", "x"),
+      None
+    )(replyQueue)
+    val (jmsRrClient, closer) = clientRes.allocated.unsafeRunSync()
+    complete {
+      super.withFixture(test.toNoArgAsyncTest(jmsRrClient))
+    } lastly {
+      closer.unsafeRunSync()
+    }
+  }
+
+  "SQS Client" - {
+    "send a message and handle the reply" in { jmsRrClient =>
+      val requestQueue = "request-general"
+      // The following block can be used in a function within the service layer of a Play Framework application.
+      // It works by creating a Queue and putting it into the ReplyMessageHandler. It next makes a request using the JMS
+      // client and when the reply eventually appears in the queue in can be returned to the caller of the service
+      // function (e.g. a Play controller) where it can be converted to a Future[Result].
+      val result = Queue.bounded[IO, String](1).flatMap { editSetQueue =>
+        val replyHandler: ReplyMessageHandler[IO] = replyMessage => editSetQueue.offer(replyMessage.body)
+        jmsRrClient.request(requestQueue, RequestMessage(s"hello 1234", "1234"), replyHandler) flatMap { _ =>
+          editSetQueue.take
+        }
+      }
+
+      IO(result.unsafeRunSync()).asserting(_ mustBe "Echo Server: hello 1234")
+    }
+    "send two messages and handle the replies" in { jmsRrClient =>
+      val requestQueue = "request-general"
+
+      // The following block can be used in a function within the service layer of a Play Framework application.
+      // It works by creating a Queue and putting it into the ReplyMessageHandler. It next makes a request using the JMS
+      // client and when the reply eventually appears in the queue in can be returned to the caller of the service
+      // function (e.g. a Play controller) where it can be converted to a Future[Result].
+      val result1 = Queue.bounded[IO, String](1).flatMap { editSetQueue =>
+        val replyHandler: ReplyMessageHandler[IO] = replyMessage => editSetQueue.offer(replyMessage.body)
+        jmsRrClient.request(requestQueue, RequestMessage(s"hello 1234", "1234"), replyHandler) flatMap { _ =>
+          editSetQueue.take
+        }
+      }
+      val result2 = Queue.bounded[IO, String](1).flatMap { editSetQueue =>
+        val replyHandler: ReplyMessageHandler[IO] = replyMessage => editSetQueue.offer(replyMessage.body)
+        jmsRrClient.request(requestQueue, RequestMessage(s"hello 5678", "5678"), replyHandler) flatMap { _ =>
+          editSetQueue.take
+        }
+      }
+
+      IO(result1.unsafeRunSync()).asserting(_ mustBe "Echo Server: hello 1234")
+      IO(result2.unsafeRunSync()).asserting(_ mustBe "Echo Server: hello 5678")
+
+    }
+    "send three messages and handle the replies" in { jmsRrClient =>
```

Comment:
Why are the sids different? Would this work if they were the same?

Reply:
See above.